### DISCUSSION TITLE
Slow when running on multiple threads

### DISCUSSION THREAD

**Discussion by mrkevinhsiao** [abuzuhri/Amazon-SP-API-CSharp#261] [General]
I have roughly 2000 Amazon accounts that I am getting the orders from.

I am doing in batches (~20 accounts), all in separate threads.

MaxThrottledRetryCount has been set to 0 and I am not getting any errors, but it is extremely slow compare to doing 1 account at a time.

Is anyone having this issue as well?

I am using this NuGet package in production and the performance is killing us at the moment.

We suspect the async threads could be hanging or delayed or in asleep.

Is it possible to tell it not to run asynchronously?

**Comment by abuzuhri** [comment]
Can you share the code use i need to take look ...

its shouldn't be slow but i need to take look to your code if possible

**Comment by abuzuhri** [comment]
after PR #270 I think now problem should be solved
